framework/core/js/src/forum/components/UserSecurityPage.tsx

```diff
@@ -51,17 +51,27 @@ export default class UserSecurityPage<CustomAttrs extends IUserPageAttrs = IUser
   settingsItems() {
     const items = new ItemList<Mithril.Children>();
 
-    ['developerTokens', 'sessions'].forEach((section) => {
-      const sectionName = `${section}Items` as 'developerTokensItems' | 'sessionsItems';
-      const sectionLocale = camelCaseToSnakeCase(section);
-
+    if (
+      app.forum.attribute('canCreateAccessToken') ||
+      app.forum.attribute('canModerateAccessTokens') ||
+      (this.state.hasLoadedTokens() && this.state.getDeveloperTokens()?.length)
+    ) {
       items.add(
-        section,
-        <FieldSet className={`UserSecurityPage-${section}`} label={app.translator.trans(`core.forum.security.${sectionLocale}_heading`)}>
-          {this[sectionName]().toArray()}
+        'developerTokensItems',
+        <FieldSet className="UserSecurityPage-developerTokens" label={app.translator.trans(`core.forum.security.developer_tokens_heading`)}>
+          {this.developerTokensItems().toArray()}
         </FieldSet>
       );
-    });
+    } else if (!this.state.hasLoadedTokens()) {
+      items.add('developerTokensItems', <LoadingIndicator />);
+    }
```

Comment:
This should not be necessary when it wasn't before, the loading indicator is shown at the list level.
```suggestion
```

Reply:
Ah I see what you mean now. That makes sense, sorry for the confusion.